tests/cecil-tests/GetterExceptionTest.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using Mono.Cecil;
+using System.Collections.Generic;
+using NUnit.Framework;
+using System.Linq;
+using Mono.Cecil.Cil;
+using Xamarin.Utils;
+using Xamarin.Tests;
+
+namespace Cecil.Tests {
+	[TestFixture]
+	public class GetterExceptionTest {
+		readonly HashSet<string> exceptionsToSkip = new ()
+		{
+			"AudioQueueException",
+			"ModelNotImplementedException",
+			"You_Should_Not_Call_base_In_This_Method"
+		};
+
+		bool IsMemberObsolete (ICustomAttributeProvider member)
+		{
+			if (member is null || !member.HasCustomAttributes)
+				return false;
+
+			return member.CustomAttributes.Any ((m) =>
+					m.AttributeType.Name == "ObsoleteAttribute" ||
+					m.AttributeType.Name == "AdviceAttribute" ||
+					m.AttributeType.Name == "ObsoletedOSPlatformAttribute");
+		}
```

Comment:
I believe we can write everything in a single line, member?.CustomAttributes?.Any() == true should do the trick. If member is null and custom attrs are null, null != true. Any will only be executed if CustomAttrs is not null.
```suggestion
		bool IsMemberObsolete (ICustomAttributeProvider member)
		    => member?.CustomAttributes?.Any ((m) =>
					m.AttributeType.Name == "ObsoleteAttribute" ||
					m.AttributeType.Name == "AdviceAttribute" ||
					m.AttributeType.Name == "ObsoletedOSPlatformAttribute") == true; // can be null or false. And null != true
```

Reply:
```csharp
"ObsoletedOSPlatformAttribute") == true;
```